src/main/scala/co/teapot/graph/MemoryMappedDirectedGraphConverter.scala

```diff
@@ -0,0 +1,251 @@
+        if (dir == EdgeDir.Out)
+          distinctEdgeCount += edgeCount
+        val segmentSize = (nodesPerSegment + 1) * BytesPerNodeOffset + edgeCount * BytesPerNeighbor
+        require(segmentSize < Integer.MAX_VALUE, "Segment too large for Int indexing")
+        segmentSizes(dir)(segmentI) = segmentSize.toInt
+        log(s"wrote ${segmentI + 1} of $segmentCount $dir-segments.")
+      }
+    }
+    outputStream.close()
+  }
+
+  /**
+    * Given an array of neighbor arrays for a segment, first computes and writes offsets (relative
+    * to the start of this segment), then writes the neigbhor data.
+    */
+  private def writeEdgeDataToStream(neighborArrayLists: Array[IntArrayList], out: DataOutputStream): Unit = {
+    var offset = 0
+    for (neighborList <- neighborArrayLists) {
+      out.writeInt(offset)
+      offset += neighborList.size * BytesPerNeighbor
+    }
+    out.writeInt(offset) // Write extra offset for computing degree of last node
+
+    for (neighborList <- neighborArrayLists) {
+      for (j <- 0 until neighborList.size) {
+        out.writeInt(neighborList.get(j))
+      }
+    }
+  }
+
+  private def cleanUp(): Unit = {
+    temporaryFilesById1 foreach (_.delete())
+    temporaryFilesById2 foreach (_.delete())
+  }
+
+  private def temporaryFilesInDirection(d: EdgeDir): Array[File] = d match {
+    case EdgeDir.Out => temporaryFilesById1
+    case EdgeDir.In => temporaryFilesById2
+  }
+
+  private def createTemporaryFile(): File = {
+    val newFile = File.createTempFile("edge_partition", ".bin")
+    newFile.deleteOnExit() // Request file gets deleted in case an exception happens
+    newFile
+  }
+
+  private def createOutputStream(f: File): DataOutputStream =
+    new DataOutputStream(new BufferedOutputStream(new FileOutputStream(f)))
+
+  private def createInputStream(f: File): DataInputStream =
+    new DataInputStream(new BufferedInputStream(new FileInputStream(f)))
+
+  /** Parses each line of the given file, assumed to each have the form "<int><whitespace><int>"
+    * and calls the given function of each pair of ints.  Empty lines are ignored.
+    */
+  private def forEachEdge(edgeListFile: File)(f: (Int, Int) => Unit): Unit = {
+    val linePattern = Pattern.compile(raw"(\d+)\s+(\d+)")
+    for (line <- Source.fromFile(edgeListFile).getLines()
+         if line.nonEmpty) {
+      val matcher = linePattern.matcher(line)
+      if (!matcher.matches()) {
+        throw new IOException("invalid line in edge file: " + line)
+      } else {
+        val u = matcher.group(1).toInt // Groups are 1-indexed
+        val v = matcher.group(2).toInt
+        f(u, v)
+      }
+    }
+  }
+
+  /** Given positive integers n and k, returns a positive integer n' such that n' is a multiple
+    * of k and n <= n' < n + k. Assumes n + k < pow(2, 31). */
+  private def nextMultiple(n: Int, k: Int): Int =
+    (n + k - 1) / k * k
+}
+
+object EdgeDir extends Enumeration {
+  type EdgeDir = Value
+  val Out, In = Value
+}
```

Comment:
Sure, I just made one at
https://github.com/plofgren/bidirectional-random-walk/pull/6
It's hard to align exactly, because I've refactored things into more modular methods, but I was able to get two parts to align.